### PR TITLE
[tado] Prepare for migration to bnd build

### PR DESCRIPTION
**Pull request by Hilbrand**
This pr contains 2 changes:
1) Adapted generated api library to generate a OSGi compatible library. This binding contains a jar in lib that is generated from swagger. The jar is generated with a pom part of this project. This pom has been adapted to generate an OSGi compatible jar to make migrating to bnd build more easily. For this the version of this jar is set to 1.2 **Note:** It might be good to check if the changes I made are correct.
2) Removed dependency on google guava library.

### CODE REVIEW

File: addons/binding/org.openhab.binding.tado/src/main/api/pom.xml [hunk 1]
```diff
@@ -55,11 +56,29 @@
 				</dependencies>
 			</plugin>
 
+			<plugin>
```
Comment (J-N-K) [inline addons/binding/org.openhab.binding.tado/src/main/api/pom.xml]: That's a bit unexpected. Why is this part changed?

Comment (Hilbrand) [reply addons/binding/org.openhab.binding.tado/src/main/api/pom.xml]: This pom is used to generate the jar in the lib folder in this project. This change allows for generating of a jar version that is OSGi compatible so no need to wrap the library. This change includes a newly generated jar file from this changed pom.
